src/pointer.cr

```diff
@@ -129,6 +129,16 @@ struct Pointer(T)
     (self + offset).value = value
   end
 
+  {% if flag?(:x86_64) %}
+    private def to_size_t(size : Int) : UInt64
+        size.to_u64
+    end
+  {% else %}
+    private def to_size_t(size : Int) : UInt32
+        size.to_u32
+    end
+  {% end %}
+
```

Comment:
No idea what you're talking about. Why spray the definition of size_t all over the codebase when it already exists.
Use `LibC::SizeT.new` instead of `to_size_t` and you don't need this method.